packages/fxa-settings/src/components/Settings/index.tsx

```diff
@@ -39,6 +46,16 @@ export const Settings = (_: RouteComponentProps) => {
     config.metrics.navTiming.endpoint,
   ]);
 
+  useEffect(() => {
+    function handleWindowFocus() {
+      if (session.isDestroyed) {
+        hardNavigate('/');
```

Comment:
I think we should use `hardNavigateToContentServer` so it's clear we need to change this when email-first is converted, and I think we're going to want to pass the query params as well. But, this is a really interesting/nice solution! 🙂 Testing now.